src/workerd/api/streams/standard.c++

```diff
@@ -3333,6 +3236,18 @@ kj::Promise<DeferredProxy<void>> ReadableStreamJsController::pumpTo(
   KJ_REQUIRE(!isLockedToReader(), "This ReadableStream is currently locked to a reader.");
   disturbed = true;
 
+  // This operation will leave the ReadableStream locked and disturbed. It will consume
+  // the stream until it either closed or errors. If the deferred proxy promise or it's
+  // inner promise is dropped, the PumpToReader (and sink) will be dropped and the stream
+  // will be canceled. If the PumpToReader is dropped while there is a pending write on
+  // the sink, the pending write will be canceled.
+
+  const auto handlePump = [&] {
+    KJ_ASSERT(lock.lock());
+    auto reader = kj::heap<PumpToReader>(addRef(), kj::mv(sink), end);
+    return addNoopDeferredProxy(reader->pumpTo(js).attach(kj::mv(reader)));
```

Comment:
If I interpret the comment for handlePump() correctly the doClose() can be dropped here becauce the stream is being closed in pumpTo(), right?

Reply:
Yes, that's it exactly